pkg/metrics/metrics.go

```diff
@@ -44,12 +46,18 @@ func NewMetrics() Metrics {
 			Help: "completed pipelines",
 		},
 			[]string{"result"}),
-		Duration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
-			Name:    "steward_pipelinerun_duration_seconds",
-			Help:    "pipeline run durations",
+		TotalDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "steward_pipelinerun_total_duration_seconds",
+			Help:    "pipeline run durations after they changed their status",
 			Buckets: prometheus.ExponentialBuckets(0.125, 2, 15),
 		},
 			[]string{"state"}),
+		CurrentDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "steward_pipelinerun_current_duration_seconds",
+			Help:    "pipeline run durations in their current status",
+			Buckets: prometheus.ExponentialBuckets(60, 2, 7),
+		},
```

Comment:
I find the terms "total" vs. "current" hard to understand. What about "finished" vs. "ongoing"?

Reply:
Renamed to steward_pipelinerun_ongoing_status_duration_seconds and steward_pipelinerun_finished_status_duration_seconds. Its quite long but I dont want that it sounds like we only meter finished runs

Reply:
A very important difference between these two metrics is that "ongoing" counts same instances multiple times, while "finished" does not. This means the metrics must be interpreted differently. However, the naming doesn't reflect this important difference but falsely suggests symmetry.

I propose the following names:

- `steward_pipelinerun_state_duration_seconds` for the histogram of pipline run states by duration. I intentionally left out "finished" or similar, because from the description it is expected that only finished stated are counted (a duration can be known only after the fact).

- `steward_pipelinerun_ongoing_state_duration_periodic_observations_seconds` for the histogram of ongoing pipeline run states by periodically observed duration. It's a rather long name, but it's also a non-trivial metric.

    In addition, the metric help text should extended to explain what exactly the numbers mean, including the possibility of multiple counting.